component/org.wso2.carbon.identity.oauth2.grant.token.exchange/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/utils/TokenExchangeUtils.java

```diff
@@ -530,8 +530,13 @@ private static boolean isJWKSEnabled() {
     private static boolean validateUsingJWKSUri(SignedJWT signedJWT, String jwksUri) throws IdentityOAuth2Exception {
 
         JWKSBasedJWTValidator jwksBasedJWTValidator = new JWKSBasedJWTValidator();
-        return jwksBasedJWTValidator.validateSignature(signedJWT.getParsedString(), jwksUri,
-                signedJWT.getHeader().getAlgorithm().getName(), null);
+        try {
+            return jwksBasedJWTValidator.validateSignature(signedJWT.getParsedString(), jwksUri,
+                    signedJWT.getHeader().getAlgorithm().getName(), null);
+        } catch (IdentityOAuth2Exception e) {
+            log.error("Error occurred while validating the signature using JWKS URI.", e);
```

Comment:
This is a client error and ideally, we shouldn't log these as ERRORs. The reason is someone can send invalid jwts and fill up logs. 

We can make this a debug log and then send back a proper error response indicating the error to the client (if its not happening now.)

Was there any specific need to print an error log for this?

Reply:
yeah, makes sense.
https://github.com/wso2-support/identity-inbound-auth-oauth/blob/3568b25aeac996d64c090972ca7cdcb3cc52e25f/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/jwt/JWKSBasedJWTValidator.java#L93
We saw this error log. We needed to capture the same thing as there was an intermittent failure from asgardeo JWKS endpoint, so that we can see it in our application logs. 
In the particular dependency, we have been using in choreo-product-apim doesn't have the fore-mentioned log. So would it be okay to add it there?

Reply:
Agreed on the fact that, it would create so many logs if the JWT is invalid.